godot-macros/src/docs.rs

```diff
@@ -135,13 +136,66 @@ fn xml_escape(value: String) -> String {
     result
 }
 
-/// Calls [`extract_docs_from_attributes`] and converts the result to BBCode for Godot's consumption.
-fn attribute_docs_to_bbcode(doc: &[venial::Attribute]) -> Option<String> {
-    let doc = extract_docs_from_attributes(doc)
-        .collect::<Vec<String>>()
-        .join("\n");
+/// Returns user docs with its `@deprecated` and `@experimental` attributes extracted.
+fn docs_with_attributes(doc: &[venial::Attribute]) -> (Vec<String>, Vec<String>, Vec<String>) {
+    let (mut docs, mut deprecated, mut experimental): (Vec<String>, Vec<String>, Vec<String>) =
+        (Vec::new(), Vec::new(), Vec::new());
 
-    (!doc.is_empty()).then(|| markdown_converter::to_bbcode(&doc))
+    let docs_ptr = &docs as *const _;
```

Comment:
Please avoid `as`  casts and use [`ptr::from_ref()` ](https://doc.rust-lang.org/std/ptr/fn.from_ref.html) instead.

Also, a comment might help here since the "select the bucket to write" is a very unusual pattern for pointers.